dyno-jedis/src/main/java/com/netflix/dyno/jedis/DynoJedisPipeline.java

```diff
@@ -61,63 +62,120 @@
+            try {
+                connection = connPool.getConnectionForOperation(new BaseOperation<Jedis, String>() {
 
-        if (theKey.get() != null) {
-            verifyKey(key);
+                    @Override
+                    public String getName() {
+                        return DynoPipeline;
+                    }
 
-        } else {
+                    @Override
+                    public String getKey() {
+                        return key;
+                    }
 
-            boolean success = theKey.compareAndSet(null, key);
+                });
+            } catch (NoAvailableHostsException nahe) {
+                cpMonitor.incOperationFailure(connection != null ? connection.getHost() : null, nahe);
+                discardPipelineAndReleaseConnection();
+                throw nahe;
+            }
+        } catch (NoAvailableHostsException nahe) {
+            cpMonitor.incOperationFailure(connection != null ? connection.getHost() : null, nahe);
+            discardPipelineAndReleaseConnection();
+            throw nahe;
+        }
+        Jedis jedis = ((JedisConnection) connection).getClient();
+        jedisPipeline = jedis.pipelined();
+        cpMonitor.incOperationSuccess(connection.getHost(), 0);
+    }
+
+    private void checkHashtag(final String key, final String hashtagValue) {
+        if (this.hashtag.get() != null) {
+            verifyHashtagValue(hashtagValue);
+        } else {
+            boolean success = this.hashtag.compareAndSet(null, hashtagValue);
             if (!success) {
-                // someone already beat us to it. that's fine, just verify that the key is the same
-                verifyKey(key);
+                verifyHashtagValue(hashtagValue);
             } else {
+                pipelined(key);
+            }
+        }
 
-                try {
-                    connection = connPool.getConnectionForOperation(new BaseOperation<Jedis, String>() {
+    }
 
-                        @Override
-                        public String getName() {
-                            return DynoPipeline;
-                        }
+    /**
+     * Checks that a pipeline is associated with a single key. If there is a
+     * hashtag defined in the first host of the connectionpool then we check
+     * that first.
+     * 
+     * @param key
+     */
+    private void checkKey(final String key) {
 
-                        @Override
-                        public String getKey() {
-                            return key;
-                        }
-                    });
-                } catch (NoAvailableHostsException nahe) {
-                    cpMonitor.incOperationFailure(connection != null ? connection.getHost() : null, nahe);
-                    discardPipelineAndReleaseConnection();
-                    throw nahe;
+        // Get hashtag from the first host of the active pool
+        String hashtag = connPool.getActivePools().iterator().next().getHost().getHashtag();
```

Comment:
@shailesh33 @jcacciatore To get the hashtag we select the first one from the active pool. We work under the assumption that all nodes will contain the same hashtag.